File: backend/app/schemas/resume.py
from datetime import datetime
from typing import List

from pydantic import AnyHttpUrl, BaseModel, Field


class JobDescriptionInput(BaseModel):
    role_title: str = Field(..., max_length=255)
    company_name: str | None = Field(default=None, max_length=255)
    canonical_text: str | None = None
    required_skills: List[str] = Field(default_factory=list)
    preferred_skills: List[str] = Field(default_factory=list)


class ResumeCreate(BaseModel):
    owner_id: int
    file_url: AnyHttpUrl
    parsed_text: str | None = None
    extracted_skills: List[str] = Field(default_factory=list)
    extracted_keywords: List[str] = Field(default_factory=list)


class ResumeRead(ResumeCreate):
    id: int
    created_at: datetime
    updated_at: datetime


class ScoreCardResponse(BaseModel):
    resume_id: int
    job_description_id: int | None
    ats_score: float
    keyword_match: float
    formatting_score: float
    overall_score: float
    recommendations: List[str]


class ResumeScoreRequest(BaseModel):
    resume: ResumeCreate
    job_description: JobDescriptionInput | None = None


class ResumeScoreResponse(BaseModel):
    resume: ResumeRead
    score_card: ScoreCardResponse


class ResumeUploadResponse(BaseModel):
    file_url: AnyHttpUrl

File: frontend/src/types/resume.ts
export type Resume = {
  id: number;
  owner_id: number;
  file_url: string;
  parsed_text?: string | null;
  extracted_skills: string[];
  extracted_keywords: string[];
  created_at: string;
  updated_at: string;
};

export type JobDescription = {
  role_title: string;
  company_name?: string | null;
  canonical_text?: string | null;
  required_skills: string[];
  preferred_skills: string[];
};

export type ScoreCard = {
  resume_id: number;
  job_description_id: number | null;
  ats_score: number;
  keyword_match: number;
  formatting_score: number;
  overall_score: number;
  recommendations: string[];
};

export type ResumeScoreRequest = {
  resume: {
    owner_id: number;
    file_url: string;
    parsed_text?: string | null;
    extracted_skills: string[];
    extracted_keywords: string[];
  };
  job_description?: JobDescription | null;
};

export type ResumeScoreResponse = {
  resume: Resume;
  score_card: ScoreCard;
};

export type ResumeUploadResponse = {
  file_url: string;
};

